pkgs/development/libraries/libguestfs/default.nix

```diff
@@ -65,6 +65,8 @@ stdenv.mkDerivation rec {
 
   doInstallCheck = appliance != null;
   installCheckPhase = ''
+    runHook preInstallCheck
```

Comment:
please stop inserting needless `runHook`s

Reply:
I don't think they are needless. Phases overrides ignoring pre and post hooks makes for bad examples. I'm not waiting for the situation when we'll need to add a pre build task on a batch of derivations for whatever reason and then realise half of them are just ignoring it. (it's an example).
So I believe it's good hygiene to start and end with pre and post hook whenever one resort to rewriting an entire phase.

Reply:
No, it is not.
The almost only place when they are needed are function like `buildPerlModule` which accept `args` and pass this args down
https://github.com/NixOS/nixpkgs/blob/c28659e2fbe37841bf2528a79759d13bf6ce87ef/pkgs/top-level/perl-packages.nix#L45-L51

Without explicit runHook, the hooks will be lost and `buildPerlModule { preSometing = "..." }` won't work.


About overrides I have three objection:
1. overrides is an exceptional mechanism which should not require explicit support. especially which requires adding about 200.000 lines if your proposal will be implementing treewide.
2. If it really needed, it should be adding it treewide, adding it using a partisan tactic in randomly chosen dozen places is ridiculous.
3. in my opinion, hook-support are is not even needed for overrides as in override if one want to prepend/postpend code she can write
```
somederivation.overrideDerivation(old: {
installPhase = ''
  some prefix
  ${old.installPhase}
  some postfix
'';
}
```
explicit hooks only add confusions